tower/src/buffer/worker.rs

```diff
@@ -80,11 +80,11 @@ where
         }
 
         tracing::trace!("worker polling for next message");
-        if let Some(mut msg) = self.current_message.take() {
-            // poll_closed returns Poll::Ready is the receiver is dropped.
-            // Returning Pending means it is still alive, so we should still
-            // use it.
-            if msg.tx.poll_closed(cx).is_pending() {
+        if let Some(msg) = self.current_message.take() {
+            // If the oneshot sender is closed, then the receiver is dropped,
+            // and nobody cares about the response. If this is the case, we
+            // should continue to the next request.
+            if !msg.tx.is_closed() {
```

Comment:
n.b. that `tokio` 0.3.2 is adding back `oneshot::Sender::poll_closed`, but I don't think we actually _need_ to poll here, since we immediately consume the sender if it isn't closed. we don't need to register interest in it closing, since we're not going to yield until we're done with that sender...so `is_closed` is probably slightly more efficient, too.